challenge_01/base_solution.py

```diff
@@ -0,0 +1,103 @@
+from typing import List, Tuple
+
+import torch
+import torch.nn as nn
+import torch.utils.data as data
+
+
+class MultiLayerPerceptron(nn.Module):
+    """
+    Model class follow rules by nn.Module
+    I tried to fill class by glues I got.
+    Key trick is create hidden layers by blocks follow nn.Sequential
+    """
+
+    def __init__(
+        self,
+        layer_dim_list: List[int],
+        activation: nn.modules.activation,
+        batch_norm: bool,
+        dropout_p: float,
+    ) -> None:
+        super().__init__()
+        self.layer_dim_list = layer_dim_list
+        self.activation = activation
+        self.batch_norm = batch_norm
+        self.dropout_p = dropout_p
+        self._create_mlp()
+
+    def _create_block(self, n_in: int, n_out: int, batch_norm: bool) -> nn.Module:
+        if batch_norm:
+            block = nn.Sequential(
+                nn.Linear(n_in, n_out),
+                nn.BatchNorm1d(n_out),
+                nn.Dropout(p=self.dropout_p),
+                self.activation(),
+            )
+        else:
+            block = nn.Sequential(
+                nn.Linear(n_in, n_out), nn.Dropout(p=self.dropout_p), self.activation()
+            )
+        return block
```

Comment:
přijde mi víc standard mít 
```
blocks = [ ]

if bn:
    blocks.append()
if dropoup:
...
blocks.append(activation)
nn.Sequantial(*blocks)
```

Reply:
MLP je tak jednoduchá síť že se to většinou dává rovnou do create_mlp

Reply:
Aj mne sa zdal ten if else kostrbatý dik za radu je to tam ;) hej je jednoduchá ale dal si na vstup list a ten list môže byť rozne velký s roznimi parametrami, tak som to urobil modulárne